internal/test/framework/framework.go

```diff
@@ -176,6 +134,35 @@ func (f *TestFramework) Setup(vmCount int, credentials []*types.Credentials, ski
 	if err := f.getLatestCniPluginsVersion(); err != nil {
 		return fmt.Errorf("unable to get latest 0.8.x version of CNI Plugins: %v", err)
 	}
+	if err := f.getKubeAPIClient(); err != nil {
```

Comment:
is there a difference between the client here and the one at line 122?

Reply:
I have the same question. It is unclear why you need both clients.

Reply:
One is kubernetes clientset of type `kubernetes.Clientset` to query the cluster status and other is kubernetes client of type  `client.Client` to interact with kubernetes API servers

Reply:
I still don't follow. Aren't you talking to the API server in both cases?

Reply:
This should have been brought up at the stand up then